Packs/Coralogix/Integrations/Coralogix.py

```diff
@@ -0,0 +1,447 @@
+    except Exception as ex:
+        return 'Test failed (' + type(ex).__name__ + ')'
+
+    if results_raw is not None and 'hits' in results_raw:
+        return 'ok'
+    else:
+        return 'Test failed (No result was received from Coralogix or the response was unexpected `' + json.dumps(results_raw) + '`)'
+
+def fetch_incidents(cgx_private_key, cgx_endpoint_url, incidents_base_query, application_name, subsystem_name, severity, incidents_name_field):
+    """
+    This method handles querying Coralogix for incidents by using the configured query and parameters
+
+    Args:
+        cgx_private_key - String, Coralogix Private Key
+        cgx_endpoint_url - String, Coralogix Cluster Base URL for the Coralogix ES-API
+        application_name - String, The
+        subsystem_name - String,
+        severity - String
+
+    Returns:
+        Returns the incidents found in Coralogix
+    """
+
+    last_run_timestamp = -1
+    if "last_run_timestamp" in demisto.getLastRun():
+        last_run_timestamp = demisto.getLastRun()["last_run_timestamp"]
+    raw_data = clx_search_command(
+        cgx_private_key=cgx_private_key,
+        cgx_endpoint_url=cgx_endpoint_url,
+        query=incidents_base_query,
+        application_name=application_name,
+        subsystem_name=subsystem_name,
+        severity=severity,
+        exclude_columns=[],
+        results_as_table=False,
+        since_timestamp=last_run_timestamp
+    )
+
+    newest_incident_date_obj = datetime.datetime(year=1970, month=1, day=1)
+    incidents = []
+    if "hits" in raw_data and "hits" in raw_data["hits"]:
+        for document in raw_data["hits"]["hits"]:
+            flattened_document = flatten_json(document["_source"])
+            incident_date = flattened_document['@timestamp']
+            incident = {
+                'name': flattened_document[incidents_name_field],
+                'occurred': incident_date,
+                'rawJSON': json.dumps(flattened_document)
+            }
+            incidents.append(incident)
+            incident_date_obj = dateutil.parser.parse(incident_date)
+            if incident_date_obj.replace(tzinfo=datetime.timezone.utc).timestamp() > newest_incident_date_obj.replace(tzinfo=datetime.timezone.utc).timestamp():
+                newest_incident_date_obj = incident_date_obj
+
+    demisto.setLastRun({"last_run_timestamp": newest_incident_date_obj.replace(tzinfo=datetime.timezone.utc).timestamp()})
+    return incidents
+
+def main():
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+    private_key = demisto.params()['privatekey'].strip()
+
+    # get the service API url
+    tags_api_url = urljoin(demisto.params()['webapi_url'].strip(), '/api/v1/addTagPost')
+    search_api_url = urljoin(demisto.params()['esapi_url'].strip(), '/*/_search')
+
+    tag_application_name = demisto.params()['app_name'].strip()
+    tag_subsystem_name = demisto.params()['subsystem_name'].strip()
```

Comment:
Why are these parameters if you are only using it in clx_tag command?
Looks like they should be a command arguments

Reply:
In Coralogix these parameters represent which system created the tag, in this case it will be by default Cortex XSOAR. If the user would like to change it he can via the integration instance parameters